### PR TITLE
Adds SRFI 176 version flag

### PR DESCRIPTION
**Pull request by higepon**
```sh
 ./mosh --loadpath=./lib tests/srfi/version.sps 
((version "0.2.8")
 (command "mosh")
 (scheme.id mosh)
 (install-dir "/usr/local")
 (languages scheme r6rs)
 (encodings utf-8)
 (mosh.cache-path "/root/.mosh")
 (website "https://mosh.monaos.org/")
 (scheme.path "/usr/local/share/mosh-0.2.8"))
```

### Remaining issue
- I'd like to have (command "nmosh") when it is running in nmosh instead of mosh

### CODE REVIEW

File: lib/srfi/%3a176/version-flag.sls [hunk 1]
```diff
@@ -0,0 +1,16 @@
+(library (srfi :176 version-flag)
+         (export version-alist)
+         (import (rnrs) (only (mosh config) get-config))
+
+(define (version-alist)
+  `((version ,(get-config "version"))
+      (command "mosh")
+      (scheme.id mosh)
+      (install-dir ,(string-append (get-config "prefix") "/bin"))
```
Comment (lassik) [inline lib/srfi/%3a176/version-flag.sls:9]: This should probably be just `(install-dir ,(get-config "prefix"))`.

File: lib/srfi/%3a176.mosh.sls [hunk 2]
```diff
@@ -0,0 +1,11 @@
+;; this file is an alias-library.
+;;  alias of:
+;;   lib/srfi/%3a176/version-flag.sls
+(library (srfi :176)
+         (export
+             version-alist
+         )
+         (import
+             (srfi :176 version-flag)
```
Comment (lassik) [inline lib/srfi/%3a176.mosh.sls:9]: Should be `(srfi :176 version)`. I'll send a PR to add the `version` library name to [`srfi-data.scm`](https://github.com/scheme-requests-for-implementation/srfi-common/blob/master/admin/srfi-data.scm).

File: lib/srfi/%3a176/version-flag.sls [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+      (languages scheme r6rs)
+      (encodings utf-8)
+      (cache-path ,(get-config "mosh-cache-dir"))
```
Comment (lassik) [inline lib/srfi/%3a176/version-flag.sls:12]: Non-standard options (not mentioned in the SRFI) should either:

* Be prefixed with the implementation's `scheme-id`, i.e. `(mosh.cache-path ,(get-config "mosh-cache-dir"))`
* Be added to `https://registry.scheme.org/#version-flag-property`.

Is the Mosh cache-path for byte-compiled files?

File: lib/srfi/%3a176/version-flag.sls [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+      (website "https://mosh.monaos.org/")
+      (scheme.path ,(get-config "library-path"))
```
Comment (lassik) [inline lib/srfi/%3a176/version-flag.sls:14]: Does `(get-config "library-path")` always return only one directory, or can it return multiple directories?

**Comment by lassik** [comment]
> * I'd like to have (command "nmosh") when it is running in nmosh instead of mosh

Yes, that makes sense.

**Comment by lassik** [comment]
I just added `mosh` to https://registry.scheme.org/#scheme-id

**Comment by lassik** [comment]
Do you plan to add the info to `mosh -V` (`showVersion()` in C++)?

File: lib/srfi/%3a176/version-flag.sls (hunk 1 above)
Comment (higepon) [reply lib/srfi/%3a176/version-flag.sls]: Thank you for your review! Done.

File: lib/srfi/%3a176.mosh.sls (hunk 2 above)
Comment (higepon) [reply lib/srfi/%3a176.mosh.sls]: Fixed. Thanks!

File: lib/srfi/%3a176/version-flag.sls (hunk 3 above)
Comment (higepon) [reply lib/srfi/%3a176/version-flag.sls]: Fixed.

> Is the Mosh cache-path for byte-compiled files?

Yes.

File: lib/srfi/%3a176/version-flag.sls (hunk 4 above)
Comment (higepon) [reply lib/srfi/%3a176/version-flag.sls]: It returns only one directory(=Default library path).

**Comment by higepon** [comment]
> Do you plan to add the info to `mosh -V` (`showVersion()` in C++)?

Probably not for now. It's a bit tricky.

File: lib/srfi/%3a176/version-flag.sls (hunk 4 above)
Comment (lassik) [reply lib/srfi/%3a176/version-flag.sls]: What happens if you set `MOSH_LOADPATH` to more than one directory?

**Comment by lassik** [comment]
`mosh.cache-path` -> `mosh.cache-dir`? For uniformity with `(mosh-cache-dir)` and `install-dir`.

**Comment by lassik** [comment]
> > Do you plan to add the info to `mosh -V` (`showVersion()` in C++)?
> 
> Probably not for now. It's a bit tricky.

The current info comes from:

```
(get-config "version")         aka "@VERSION@"
(get-config "prefix")          aka "@prefix@"
(get-config "mosh-cache-dir")  aka (mosh-cache-dir)
(library-path)
```

Is the same information available in C++? If so, you could use something like `printf("(version \"%s\")\n", "@VERSION@");` to write the same info in `showVersion()`.

**Comment by higepon** [comment]
> `mosh.cache-path` -> `mosh.cache-dir`? For uniformity with `(mosh-cache-dir)` and `install-dir`.

Done. Thanks.

> > > Do you plan to add the info to `mosh -V` (`showVersion()` in C++)?
> > 
> > 
> > Probably not for now. It's a bit tricky.
> 
> The current info comes from:
> 
> ```
> (get-config "version")         aka "@VERSION@"
> (get-config "prefix")          aka "@prefix@"
> (get-config "mosh-cache-dir")  aka (mosh-cache-dir)
> (library-path)
> ```
> 
> Is the same information available in C++? If so, you could use something like `printf("(version \"%s\")\n", "@VERSION@");` to write the same info in `showVersion()`.

If it's okay to show subset of (version-alist) in showVersion(), I think we can.
The data sources of ```(scheme.path "/usr/local/share/mosh-0.2.8")``` is defined in Scheme world, which is tricky.

**Comment by lassik** [comment]
Subset is OK.

**Comment by higepon** [comment]
@lassik 

It would be great if you could double check the following version-alist output(s).

### (version-alist)
scheme.path now returns multiple (and correct) library paths.
```
((version "0.2.8")
 (command "mosh")
 (scheme.id mosh)
 (install-dir "/usr/local")
 (languages scheme r6rs)
 (encodings utf-8)
 (mosh.cache-dir "/root/.mosh")
 (website "https://mosh.monaos.org/")
 (scheme.path
   "."
   "/root/mosh.git/lib"
   "/usr/local/share/mosh-0.2.8/lib"))
```

### mosh -v
```
# ./mosh -v
Mosh R6RS scheme interpreter, version 0.2.8 (mosh-0.2.8-rc3-85-gc4460bef Sun, 4 Sep 2022 20:57:44 +0900) 

(version "0.2.8")
(scheme.id mosh)
(languages scheme r6rs)
(encodings utf-8)
(website "https://mosh.monaos.org/")
```

File: lib/srfi/%3a176/version.sls [hunk 5]
```diff
@@ -0,0 +1,18 @@
+(library (srfi :176 version)
+         (export version-alist)
+         (import (rnrs)
+                 (only (mosh config) get-config)
+                 (only (mosh) library-path))
+
+(define (version-alist)
+  `((version ,(get-config "version"))
+      (command "mosh")
```
Comment (higepon) [inline lib/srfi/%3a176/version.sls:9]: @okuoku 
I want to have "nmosh" if user is running nmosh.
Is there any way to know if user is running mosh or nmosh?

Comment (okuoku) [reply lib/srfi/%3a176/version.sls:9]: Currently we have no direct way to do this so need to add some internal variable here to check https://github.com/higepon/mosh/blob/c4460bef6829262a163118b810f64528776eee3a/src/main.cpp#L296 . When mosh compiled as `nmosh`, C macro `WITH_NMOSH_DEFAULTS` will be defined.

Comment (higepon) [reply lib/srfi/%3a176/version.sls]: > . When mosh compiled as `nmosh`, C macro `WITH_NMOSH_DEFAULTS` will be defined.

Great. Thanks for the pointer. I was able to use the C macro.
CI is running now.

**Comment by lassik** [comment]
> @lassik
> 
> It would be great if you could double check the following version-alist output(s).

Everything looks good!

**Comment by higepon** [comment]
Thank you everyone!
I'm going ahead and merge this.

**Pull request merged**